### PR TITLE
Fixing errors

### LINKED ISSUE #5028: closed campaign undefinded index errors
`Notice: Undefined index: und in dosomething_campaign_run_preprocess_winners() (line 133 of /var/www/dev.dosomething.org/lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module).`

`Notice: Undefined index: und in dosomething_campaign_run_preprocess_klout_gallery() (line 153 of /var/www/dev.dosomething.org/lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module).`

http://dev.dosomething.org:8888/campaigns/closed-campaign

### LINKED ISSUE #5893: More errors on thor
image

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?

It fixes a lot of those random errors spamming our website.
#### Where should the reviewer start?

oh jeez.... you should toss the fishing line in and hope you catch something.
#### How should this be manually tested?

is there lots of spammy errors happening when you load a page?
#### Any background context you want to provide?

Most of these errors stem from a lack of isset check, bad references, etc
#### What are the relevant tickets?

Fixes #5028
Fixes #5893

### CODE REVIEW

**Comment by weerd** [comment]
checking now :fishing_pole_and_fish:

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module [hunk 1]
```diff
@@ -101,6 +101,8 @@ function dosomething_campaign_run_preprocess_winners(&$vars, $nid) {
     return;
   }
 
+  $language = dosomething_global_convert_country_to_language(dosomething_global_get_current_prefix());
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module]: This doesn't seem to get used in the function?

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module [hunk 2]
```diff
@@ -150,50 +158,56 @@ function dosomething_campaign_run_preprocess_winners(&$vars, $nid) {
...
-      $image_nid = $gallery_items[$gallery_item_entity_id]->field_gallery_image[LANGUAGE_NONE][0]['target_id'];
+  if (isset($closed_node->field_gallery_collection[LANGUAGE_NONE])) {
 
-      if ($klout_gallery[$i]['type'] == 'action') {
-        $klout_gallery[$i]['items'][$a]['image'] = dosomething_image_get_themed_image($image_nid, 'square', '400x400');
-      }
-      else if ($klout_gallery[$i]['type'] == 'press'){
-        // Smaller image for these.
-        $klout_gallery[$i]['items'][$a]['image'] = dosomething_image_get_themed_image($image_nid, 'thumb', 'wmax-300-hmax-75');
+    $klout_count =  count($closed_node->field_gallery_collection[LANGUAGE_NONE]);
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module]: :space_invader: between `=` and `count` :nitpick:

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module (hunk 1 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module]: correct, leftover ill get rid of

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module (hunk 2 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module]: This is how the code originally was and I'm not sure what you mean

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module]: 🐣⛏

File: lib/modules/dosomething/dosomething_fact/dosomething_fact.module [hunk 3]
```diff
@@ -185,7 +185,7 @@ function dosomething_fact_get_formatted_sources($sources) {
     return NULL;
   }
 
-  if (!is_array($sources[0])) {
+  if (array_key_exists("raw", $sources)) {
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_fact/dosomething_fact.module:188]: Does this bug out if you don't pass it an array?

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_fact/dosomething_fact.module:188]: just explained to diego but ill try to do it here

The problem is the arrays we're checking for could have a 0, 1, 2 etc normal array structure, or JUST be formatted and raw. 

For arrays with formatted and raw, searching for [0] tosses an error. This performs the same check without an error.

Comment (weerd) [inline lib/modules/dosomething/dosomething_fact/dosomething_fact.module:188]: :+1:

**Comment by weerd** [comment]
caught a couple :fish: but overall :+1:

**Pull request merged**